### ISSUE TITLE
Wrong type assumption with subclassed numpy.ndarray

### ISSUE THREAD

**Issue by mfbehrens99** [python/mypy#13839]
**Bug Report**

When running the following program, mypy throws a false positive error (as far as I think)

**To Reproduce**

```python

from numpy import array, ndarray


class Wrench(ndarray):
    """A 2D force vector combined with a 1D torque vector."""

    def __new__(cls, vector: ndarray) -> "Wrench":
        """
        Create a new Wrench.

        Args:
            vector: 3D vector of the wrench
        """
        return vector.view(cls)


if __name__ == "__main__":
    w = Wrench(array([3.0, 4.0, 5.0]))
    x = Wrench(array([3.0, 4.0, 5.0]))
    print(type(w + x))
    y: Wrench = w + x

```

**Expected Behavior**

Do not throw an error

**Actual Behavior**

Incompatible types in assignment (expression has type "ndarray[Any, dtype[bool_]]", variable has type "Wrench") mypy(error)

**Your Environment**

- Mypy version used: mypy 0.961 (compiled: yes)
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: Python 3.10.5

**Comment by JelleZijlstra** [comment]
Do you have a stubs package installed for numpy? This is more likely a bug in such stubs than in mypy.

**Comment by erictraut** [comment]
Mypy is doing the right thing in this case. The `ndarray.__add__` method doesn't return a `Self` type. It always returns an `ndarray`, which is not compatible with a subclass such as `Wrench`. Recommend closing.
